lane.py

```diff
@@ -387,897 +372,796 @@ def has_next_page(self):
-            # from parent lanes. This is a placeholder.
-            sublane.set_customlist_information(None, None, None)
+            # Get a new list of books that meet our (possibly newly 
+            # reduced) standards.
+            new_books = self.random_sample(query, target_size)
+            for book in new_books:
+                if book.works_id not in book_ids:
+                    books.append(book)
+                    book_ids.add(book.works_id)
+            if len(books) >= target_size:
+                # We found enough books.
+                break
+        return books[:target_size]
 
-        self.set_from_parent(
-            'list_seen_in_previous_days', list_seen_in_previous_days)
+    @classmethod
+    def featured_collection_facets(cls):
+        """Yield a sequence of Facets settings to use when trying to find
+        featured works.
 
-    def custom_lists_for_identifier(self, list_data_source, list_identifier):
-        """Turn a data source and an identifier into a specific list
-        of CustomLists.
+        The sequence represents our becoming more and more desperate
+        to find enough books to fill a 'featured' lane.
         """
-        if isinstance(list_data_source, basestring):
-            list_data_source = DataSource.lookup(self._db, list_data_source)
-        # The lane may be restricted to books that are on one or
-        # more specific lists.
-        if not list_identifier:
-            lists = None
-        elif isinstance(list_identifier, CustomList):
-            lists = [list_identifier]
-        elif (isinstance(list_identifier, list) and
-              isinstance(list_identifier[0], CustomList)):
-            lists = list_identifier
-        else:
-            if isinstance(list_identifier, basestring):
-                list_identifiers = [list_identifier]
-            q = self._db.query(CustomList).filter(
-                CustomList.foreign_identifier.in_(list_identifiers))
-            if list_data_source:
-                q = q.filter(CustomList.data_source==list_data_source)
-            lists = q.all()
-            if not lists:
-                raise UndefinedLane(
-                    "Could not find any matching lists: %s, %r" %
-                    (list_data_source, list_identifiers)
-                )
-        if list_data_source:
-            list_data_source_id = list_data_source.id
+        # The 'featured' collection contains high quality works.
+        yield (Facets.COLLECTION_FEATURED, Facets.AVAILABLE_NOW, False)
+        yield (Facets.COLLECTION_FEATURED, Facets.AVAILABLE_ALL, False)
+
+        # The 'main' collection contains everything except low-quality
+        # open-access works, 
+        yield (Facets.COLLECTION_MAIN, Facets.AVAILABLE_NOW, False)
+        yield (Facets.COLLECTION_MAIN, Facets.AVAILABLE_ALL, False)
+
+        # The 'full' collection contains absolutely everything.
+        yield (Facets.COLLECTION_FULL, Facets.AVAILABLE_ALL, False)
```

Comment:
Why is featured always false in these? It seems like you could start with true and change to false if you get more desperate. Are you thinking that's something that will be configurable when you add a list to the lane?

Reply:
WorkLists in general don't use CustomLists, so all that code is in Lane. I've added an explanation of this to the docstring, but I could also move the Lane code into here if I also move `Lane.customlists`. What do you think?